gmso/parameterization/parameterize.py

```diff
@@ -12,63 +12,47 @@ def apply(
     forcefields,
     match_ff_by="molecule",
     identify_connections=False,
-    identify_connected_components=True,
-    use_molecule_info=False,
-    assert_bond_params=True,
-    assert_angle_params=True,
-    assert_dihedral_params=True,
-    assert_improper_params=False,
+    speedup_by_molgraph=False,
+    speedup_by_moltag=False,
+    ignore_params=["improper"],
     remove_untyped=False,
     fast_copy=True,
 ):
     """Set Topology parameter types from GMSO ForceFields.
 
     Parameters
     ----------
-    top: gmso.core.topology.Topology, required
+    top : gmso.core.topology.Topology, required
         The GMSO topology on which to apply forcefields
 
-    forcefields: ForceField or dict, required
+    forcefields : ForceField or dict, required
         The forcefield to apply. If a dictionary is used the keys are labels that match
         the molecule name (specified as a label of site), and the values are gmso ForceField objects that gets applied
         to the specified molecule.
         Note: if a Topology with no molecule is provided, this option will only take
         a ForceField object. If a dictionary of ForceFields is provided, this method will
         fail.
 
-    match_ff_by: str, optional, default="molecule"
+    match_ff_by : str, optional, default="molecule"
         They site's tag used to match the forcefields provided above to the Topology.
         Options include "molecule" and "group". This option is only valid if forcefields are provided
         as a dict.
 
-    identify_connections: bool, optional, default=False
+    identify_connections : bool, optional, default=False
         If true, add connections identified using networkx graph matching to match
         the topology's bonding graph to smaller sub-graphs that correspond to an angle,
         dihedral, improper etc
 
-    identify_connected_components: bool, optional, default=True
+    speedup_by_molgraph: bool, optional, default=True
```

Comment:
I like this parameter name more, and the doc strings do a better job of explaining what this does, but do you think we should add a bit more context about when this parameter might want to be used and when it shouldn't?

Also we should change the `default` here to `False` since the parameter was changed to `False`

Reply:
I will elaborate the docstring more